### DISCUSSION TITLE
Change names of repository injection tokens to prevent overlapping with @nestjs/typeorm

### DISCUSSION THREAD

**Discussion by okovpashko** [mikro-orm/nestjs#106] [Ideas] [answered]
Hello.

During the migration from TypeORM to MikroORM I faced a very strange issue when I injected a MikroORM repository but got an exception from the code of a TypeORM repository.

My digging ends up with the finding that both `@mikroorm/nestjs` and `@nestjs/typeorm` packages by default have the same injection tokens ([first](https://github.com/mikro-orm/nestjs/blob/master/src/mikro-orm.common.ts#L17) and [second](https://github.com/nestjs/typeorm/blob/master/lib/common/typeorm.utils.ts#L53)) for repositories: "**\<entityClassName\>Repository**". 
In my case it caused the Nest.js DI container to inject a TypeORM repository instead of a MikroORM repository despite using the `@InjectRepository` decorator from the `@mikroorm/nestjs` package.

Having the assumption that MikroORM is a popular replacement for a TypeORM and also that it's not that easy to replace ORM in the whole project at a time it can be a pretty popular issue.

Long story short: can you please consider changing the injection tokens for MikroORM repositories? BTW using Symbol as an injection token would have prevented this issue but I'm not sure if it's can be used for the minimal supported Node.js version.

**Comment by B4nan** [comment, marked as answer]
> pretty popular issue.

You are the first one reporting it, so probably not that popular :]

> Long story short: can you please consider changing the injection tokens for MiktoORM?

Yeah, I guess this won't hurt anybody, but I will keep this for v6 as it might be seen as breaking.

**Reply by okovpashko** [reply]
Wow, thank you for such a quick reply!

> You are the first one reporting it, so probably not that popular :]

I didn't find the related issues or discussions and I was surprised because of that. Probably that's because people don't tend to replace the ORM on the live projects so often. Or they abandon their tries after facing issues like that. It was quite not obvious and took me more than an hour of debugging and digging in the source code to figure out the reason.
The only reason I did that is that after discovering MikroORM I understood that it solves many of my pains with TypeORM and I really wanted to try it in my project.

> Yeah, I guess this won't hurt anybody, but I will keep this for v6 as it might be seen as breaking.

Thank you!

**Reply by B4nan** [reply]
Yeah agreed it can make people surrender the migration for no real reason.
